druid/examples/calc.rs

```diff
@@ -148,6 +166,116 @@ fn flex_row<T: Data>(
     row
 }
 
+struct KeyboardHandler<T: Data> {
+    child: Flex<T>,
+}
+
+impl Widget<CalcState> for KeyboardHandler<CalcState> {
+    fn paint(
+        &mut self,
+        paint_ctx: &mut PaintCtx,
+        base_state: &BaseState,
+        data: &CalcState,
+        env: &Env,
+    ) {
+        self.child.paint(paint_ctx, base_state, data, env)
+    }
+
+    fn layout(
+        &mut self,
+        layout_ctx: &mut LayoutCtx,
+        bc: &BoxConstraints,
+        data: &CalcState,
+        env: &Env,
+    ) -> Size {
+        self.child.layout(layout_ctx, bc, data, env)
+    }
+
+    fn event(&mut self, ctx: &mut EventCtx, event: &Event, data: &mut CalcState, env: &Env) {
+        use druid::HotKey;
+        use druid::KeyCode::*;
+        use druid::RawMods;
+        use druid::RawMods::Shift;
+        use druid_shell::UnknownKeyMap;
+        use CalcInput::*;
+
+        match event {
+            Event::KeyDown(key_event) => {
+                // map to physical keys since we want numbers with numlock both on and off.
+                let key_event = key_event.map_keycode(UnknownKeyMap::to_physical);
```

Comment:
I think the thing to do here is to just avoid routing through key_event for this and avoid all the pain of `text` and `unmodified_text`.

We could perhaps implement `Eq` for `HotKey`